pkg/config/config.go

```diff
@@ -522,3 +525,47 @@ func GetSrcType(ctStr string) (SrcType, error) {
 		return NONE, fmt.Errorf("Unsupported component source type: %s", ctStr)
 	}
 }
+
+// CorrectSourcePath corrects the current sourcePath depending on local or binary configuration,
+// if Git has been passed, we simply return the source location from LocalConfig
+// this will get the correct source path whether on Windows, macOS or Linux.
+func CorrectSourcePath(localConfig *LocalConfigInfo) (path string, err error) {
+
+	cmpName := localConfig.GetName()
+	sourceType := localConfig.GetSourceType()
+	sourceLocation := localConfig.GetSourceLocation()
+
+	if sourceLocation == "" {
+		return "", fmt.Errorf("Blank source location provided")
+	}
+
+	// If we are passing in a GIT source type, we make no changes.
+	// if it's binary or local, we will correct the source path.
+	if sourceType == GIT {
+		glog.V(4).Info("Git source type detected, not correcting SourcePath location")
+		return sourceLocation, nil
+	} else if sourceType == BINARY || sourceType == LOCAL {
+
+		// We use "ToSlash" here to make sure that deliminators are in a slash format
+		sourcePath := filepath.ToSlash(localConfig.GetSourceLocation())
+		u, err := url.Parse(sourcePath)
+		if err != nil {
+			return "", errors.Wrapf(err, "unable to parse source %s from component %s", sourcePath, cmpName)
+		}
+
+		if len(u.Scheme) == 1 && runtime.GOOS == "windows" {
+			// See https://github.com/golang/go/issues/13276
+			// url/parse is unable to parse Windows file paths.. so here we detect if there is one character in the Scheme
+			// Ex. C or Z drive, etc.. as well as check to see if we're on Windows.
+			// else
+			// we check the schema normally (on Linux / macOS)
+			glog.V(4).Info("We're using Windows here and unfortunately net/url doesn't support parsing.. we're going to skip this")
+		} else if u.Scheme != "" && u.Scheme != "file" {
```

Comment:
not sure about this, do we even need this check?

Reply:
Yes, it's nice to have this check in to make sure that it's been parsed correctly and it's a file not an https location, this would be a form of validation check for `local` and `binary` to make sure that a URL hasn't been passed in.